pkg/minikube/bootstrapper/bsutil/kverify/kverify.go

```diff
@@ -44,86 +34,31 @@ import (
 	"k8s.io/minikube/pkg/minikube/logs"
 )
 
-// minLogCheckTime how long to wait before spamming error logs to console
-const minLogCheckTime = 60 * time.Second
-
-// WaitForAPIServerProcess waits for api server to be healthy returns error if it doesn't
-func WaitForAPIServerProcess(r cruntime.Manager, bs bootstrapper.Bootstrapper, cfg config.ClusterConfig, cr command.Runner, start time.Time, timeout time.Duration) error {
-	glog.Infof("waiting for apiserver process to appear ...")
-	err := wait.PollImmediate(time.Millisecond*500, timeout, func() (bool, error) {
-		if time.Since(start) > timeout {
-			return false, fmt.Errorf("cluster wait timed out during process check")
-		}
-
-		if time.Since(start) > minLogCheckTime {
-			announceProblems(r, bs, cfg, cr)
-			time.Sleep(kconst.APICallRetryInterval * 5)
-		}
-
-		if _, ierr := apiServerPID(cr); ierr != nil {
-			return false, nil
-		}
-
-		return true, nil
-	})
-	if err != nil {
-		return fmt.Errorf("apiserver process never appeared")
-	}
-	glog.Infof("duration metric: took %s to wait for apiserver process to appear ...", time.Since(start))
-	return nil
-}
+const (
+	// minLogCheckTime how long to wait before spamming error logs to console
+	minLogCheckTime = 30 * time.Second
+	// APIServerWait is the name used in the flags for k8s api server
+	APIServerWait = "apiserver"
+	// SystemPodsWait is the name used in the flags for pods in the kube system
+	SystemPodsWait = "system_pods"
+	// DefaultServiceAccountWait is the name used in the flags for default service account
+	DefaultServiceAccountWait = "default_sa"
+)
 
-// apiServerPID returns our best guess to the apiserver pid
-func apiServerPID(cr command.Runner) (int, error) {
-	rr, err := cr.RunCmd(exec.Command("sudo", "pgrep", "-xnf", "kube-apiserver.*minikube.*"))
-	if err != nil {
-		return 0, err
-	}
-	s := strings.TrimSpace(rr.Stdout.String())
-	return strconv.Atoi(s)
-}
+// DefaultWaits is map of the the default components to wait for
+var DefaultWaits = map[string]bool{APIServerWait: true, SystemPodsWait: true}
 
-// ExpectedComponentsRunning returns whether or not all expected components are running
-func ExpectedComponentsRunning(cs *kubernetes.Clientset) error {
-	expected := []string{
-		"kube-dns", // coredns
-		"etcd",
-		"kube-apiserver",
-		"kube-controller-manager",
-		"kube-proxy",
-		"kube-scheduler",
-	}
+// DefaultWaitsKeys is list of all default components to wait for
+var DefaultWaitsKeys = []string{APIServerWait, SystemPodsWait}
 
-	found := map[string]bool{}
+// NoWaitsCompos is map of componets to wait for if specified 'none' or 'false'
+var NoWaitsCompos = map[string]bool{APIServerWait: false, SystemPodsWait: false, DefaultServiceAccountWait: false}
```

Comment:
Name this "NoWaitComponents".

Also, shouldn't this just be: map[string]bool{}

Reply:
in the future we might have something super small to verify, even if they say no wait.
I find this more clear and cleaner to be explicit. plus makes the callers of this code look nicer too